ContentFiles/ElementPackages/StandardLibrary/Path.ele

```diff
@@ -0,0 +1,158 @@
+constraint PathFunc(u:Num):Vector3
+struct Bounds(lower:Num, upper:Num)
+
+
+struct Path(at:PathFunc, domain:Bounds)
+{
+    # A Path is defined by a parametric equation for a path,
+    # and a domain over which the function is valid.
+
+    constraint ScalarTransform(x:Num):Num
+    constraint Vector3Unary(v:Vector3):Vector3
+    constraint Vector3Binary(v1:Vector3, v2:Vector3):Vector3
+
+    changeDomain(this:Path, newDomain:Bounds):Path
+    {
+        # Change the path domain from the current domain "newDomain".
+        #
+        # This function returns a new path "newPath" such that
+        #
+        #   newPath.domain = newDomain
+        #   newPath.at(newDomain.lower) = this.at(this.domain.lower)
+        #   newPath.at(newDomain.upper) = this.at(this.domain.lower)
```

Comment:
I'm not sure if this description makes things clearer or not. What you are saying here is that the new domain maps onto the old, but the use of `# newPath.at(newDomain.lower) = this.at(this.domain.lower)` and `# newPath.at(newDomain.upper) = this.at(this.domain.lower)` in the description to specify that obfuscates that somewhat (to me at least), since the remapping should be transparent to the user.

Reply:
Hmm. I think it's valuable to include those examples because it makes it explicit that there's a difference between
```newPath = path.changeDomain(newDomain)```
and
```newPath = Path(path.at, newDomain)```

How about changing it to
```

        # Change the path domain from the current domain "newDomain".
        #
        # This function returns a new path where the "newDomain" maps
        # onto the original domain.
        #
        # This returned path, "newPath" has the properties:
        #   newPath.domain = newDomain
        #   newPath.at(newDomain.lower) = this.at(this.domain.lower)
        #   newPath.at(newDomain.upper) = this.at(this.domain.lower)
```

?

Reply:
Fair point, I didn't realise that the other comments further down lean on these definitions too, so having them here does make thing clearer in hindsight.

Updated description looks good!